### PR TITLE
GH-1349 Add token-regenerate command

### PR DESCRIPTION
**Pull request by Tree4Free**

### CODE REVIEW

File: reposilite-backend/src/test/kotlin/com/reposilite/token/specification/AccessTokenSpecification.kt [hunk 1]
```diff
@@ -20,6 +20,7 @@ import com.reposilite.journalist.backend.AggregatedLogger
 import com.reposilite.journalist.backend.InMemoryLogger
 import com.reposilite.journalist.backend.PrintStreamLogger
 import com.reposilite.token.AccessTokenFacade
+import com.reposilite.token.AccessTokenRepository
```
Comment (joshuasing) [inline reposilite-backend/src/test/kotlin/com/reposilite/token/specification/AccessTokenSpecification.kt:23]: This import appears to be unused, is it needed?

File: reposilite-backend/src/test/kotlin/com/reposilite/token/AccessTokenFacadeTest.kt [hunk 2]
```diff
@@ -80,4 +80,16 @@ internal class AccessTokenFacadeTest : AccessTokenSpecification() {
...
+    @Test
+    fun `should regenerate token`() = runBlocking {
+        // given: an existing token and its updated version
+        val token = createToken("reposilite").accessToken
+
+        // when: token is updated
+        accessTokenFacade.regenerateAccessToken(token, "new secret")
+
+        // then: stored token should be updated
+
```
Comment (joshuasing) [inline reposilite-backend/src/test/kotlin/com/reposilite/token/AccessTokenFacadeTest.kt:92]: Consider removing this extra blank line

File: reposilite-backend/src/main/kotlin/com/reposilite/token/infrastructure/AccessTokenCommands.kt [hunk 3]
```diff
@@ -156,3 +156,30 @@ internal class RevokeCommand(private val accessTokenFacade: AccessTokenFacade) :
     }
 
 }
+
+@Command(name = "token-regenerate", description = ["Regenerate token secret"])
+internal class RegenerateCommand(private val accessTokenFacade: AccessTokenFacade): ReposiliteCommand {
+    @Parameters(index = "0", paramLabel = "<name>", description = ["Name of token to revoke"])
```
Comment (joshuasing) [inline reposilite-backend/src/main/kotlin/com/reposilite/token/infrastructure/AccessTokenCommands.kt:162]: To help keep things consistent, consider adding a blank line above this one

File: reposilite-backend/src/main/kotlin/com/reposilite/token/infrastructure/AccessTokenCommands.kt [hunk 4]
```diff
@@ -156,3 +156,30 @@ internal class RevokeCommand(private val accessTokenFacade: AccessTokenFacade) :
...
+                            context.append("Token '$name' not found")
+                        }
+                    )
+            }
+            ?: run {
+                context.status = FAILED
+                context.append("Token '$name' not found")
+            }
+    }
+}
```
Comment (joshuasing) [inline reposilite-backend/src/main/kotlin/com/reposilite/token/infrastructure/AccessTokenCommands.kt:185]: To help keep things consistent, consider adding a blank line above this bracket

File: reposilite-backend/src/integration/kotlin/com/reposilite/token/AccessTokenCommandsIntegrationTest.kt [hunk 5]
```diff
@@ -37,4 +38,18 @@ internal abstract class AccessTokenCommandsIntegrationTest : AccessTokenIntegrat
...
+        // when: user updates the token
+        val firstResult = assertOk(consoleFacade.executeCommand("token-regenerate name -s new-secret"))
+
+
+        // then: the given token is updated
+        assertEquals(SUCCEEDED, firstResult.status)
+        val secretMatch = useFacade<AccessTokenFacade>().secretMatches(useExistingToken(name).accessToken.identifier, "new-secret")
+        assertTrue(secretMatch)
+    }
 }
```
Comment (joshuasing) [inline reposilite-backend/src/integration/kotlin/com/reposilite/token/AccessTokenCommandsIntegrationTest.kt:55]: To help keep things consistent, consider adding a blank line above this bracket

File: reposilite-backend/src/integration/kotlin/com/reposilite/token/AccessTokenCommandsIntegrationTest.kt [hunk 6]
```diff
@@ -37,4 +38,18 @@ internal abstract class AccessTokenCommandsIntegrationTest : AccessTokenIntegrat
...
 
+    @Test
+    fun `should regenerate access token secret`() {
+        // given: a token
+        val (name) = useAuth("name", "secret", listOf(), routes = mapOf("/" to READ))
+
+        // when: user updates the token
+        val firstResult = assertOk(consoleFacade.executeCommand("token-regenerate name -s new-secret"))
+
+
```
Comment (joshuasing) [inline reposilite-backend/src/integration/kotlin/com/reposilite/token/AccessTokenCommandsIntegrationTest.kt:49]: Consider removing this extra blank line

**Review by joshuasing — COMMENTED** [review]
Hey, I was some spare time and had a look through your code and noticed a few inconsistencies that I thought could be resolved.

Thanks for making this pull request though, I could have used this command a while ago :)

**Comment by Tree4Free** [comment]
@joshuasing I think I corrected all the format issues you mentioned, thanks for pointing them out.

**Comment by joshuasing** [comment]
No problem, glad I could be of help :)

**Review by dzikoysk — APPROVED** [review]

**Comment by dzikoysk** [comment]
Looks good, thanks @Tree4Free for PR & @joshuasing for review! ❤️

**Pull request merged**